### PR TITLE
chore: add packagePullStatus to consolidated API

### PR DESCRIPTION
**Pull request by ashit-rath**
## Description
This is a refactor PR to support auto pull of packages on page load.
1.  adds an additional property on the consolidated API which reflects if the application needs to upgrade it's packages (`packagePullStatus`)
2. A scaffold component to be extended in EE to show the package pull status as it's called in parallel to consolidated API and is non-blocking to the app's first page load

Not this is only relevant to the edit mode of the application

PR for https://github.com/appsmithorg/appsmith-ee/pull/5745

## Automation

/ok-to-test tags="@tag.All"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/12349195312>
> Commit: 9112f8a148a1f12c3f8cb3d59606d72afb962055
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=12349195312&attempt=2" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.All`
> Spec:
> <hr>Mon, 16 Dec 2024 12:30:28 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `PackageUpgradeStatus` component for future use.
	- Added a new enumeration `PACKAGE_PULL_STATUS` with status constants.
- **Enhancements**
	- Updated `setupPageAction` to accept additional parameters for improved data handling.
	- Enhanced error handling and logic in methods related to loading application data and managing plugins.
- **Bug Fixes**
	- Adjusted the argument structure for the `setupPage` method to ensure compatibility with updated action handling.
- **Documentation**
	- Updated interfaces and method signatures to reflect new parameters and structures.

### CODE REVIEW

File: app/client/src/entities/Engine/AppEditorEngine.ts [hunk 1]
```diff
@@ -124,13 +124,19 @@ export default class AppEditorEngine extends AppEngine {
...
       unpublishedActionCollections,
       unpublishedActions,
     } = allResponses;
     const initActionsCalls = [
-      setupPageAction(toLoadPageId, true, pageWithMigratedDsl),
+      setupPageAction({
+        id: toLoadPageId,
+        isFirstLoad: true,
+        pageWithMigratedDsl,
+        packagePullStatus,
```
Comment (vsvamsi1) [inline app/client/src/entities/Engine/AppEditorEngine.ts:138]: Can you share the EE pr with me @ashit-rath .

**Review by vsvamsi1 — APPROVED** [review]

File: app/client/src/entities/Engine/AppEditorEngine.ts (hunk 1 above)
Comment (ashit-rath) [reply app/client/src/entities/Engine/AppEditorEngine.ts:138]: @vsvamsi1  https://github.com/appsmithorg/appsmith-ee/pull/5745

**Pull request merged**